extensions/ros1/launch

```diff
@@ -2,19 +2,24 @@
 
 function activate_workspace() {
     rosdir="$1"
-    setup_sh="$rosdir/setup.bash"
+    setup_sh="$rosdir/setup.sh"
 
     if [ ! -f "$setup_sh" ]; then
         echo "error: $setup_sh not found"
         exit 1
     fi
     # shellcheck disable=SC1090
-    _CATKIN_SETUP_DIR="$rosdir" source "$setup_sh"
+    _CATKIN_SETUP_DIR="$rosdir" source "$setup_sh" "${@:2}"
 }
 
 # Save off parameters, the sourced setup scripts may manipulate them.
 original_args=("$@")
 
-activate_workspace "$SNAP/opt/ros/$ROS_DISTRO"
+# activate_workspace in content shared snaps
+if [ -d "$SNAP/opt/ros/underlay_ws/opt/ros/$ROS_DISTRO" ]; then
+    activate_workspace "$SNAP/opt/ros/underlay_ws/opt/ros/$ROS_DISTRO" --local
```

Comment:
Couldn't activate_workspace check for the workspace itself?

Reply:
In those launch scripts, the workspace activation function already checks whether the workspace exists or not. However it issues an error in case it doesn't. 
This particular line concerns the workspace mounted by content-sharing. It may or may not exists depending on whether the snap is using content-sharing (and the plug is connected) or not.
I guess we could have different launch for regular vs content-sharing snaps but that would be more trouble for little benefits.